From d59eb7c7c513b3a698c8a0d608995183e51a43ec Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 13 Jun 2022 11:57:30 +0100
Subject: [PATCH] ARM: dts: bcm2711: Use the rpi-vid-decoder driver

The rpi-vid-decoder driver has been the preferred option for a while,
so make it the default.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2711-rpi-ds.dtsi | 29 ++++++++-------------------
 1 file changed, 8 insertions(+), 21 deletions(-)

diff --git a/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi b/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi
index cd5c43adc56b..5b18e8f92145 100644
--- a/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi
+++ b/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi
@@ -75,29 +75,16 @@ xhci: xhci@7e9c0000 {
 		power-domains = <&power RPI_POWER_DOMAIN_USB>;
 	};
 
-	hevc-decoder@7eb00000 {
-		compatible = "raspberrypi,rpivid-hevc-decoder";
-		reg = <0x0 0x7eb00000  0x0 0x10000>;
-		status = "okay";
-	};
-
-	rpivid-local-intc@7eb10000 {
-		compatible = "raspberrypi,rpivid-local-intc";
-		reg = <0x0 0x7eb10000  0x0 0x1000>;
-		status = "okay";
+	codec@7eb10000 {
+		compatible = "raspberrypi,rpivid-vid-decoder";
+		reg = <0x0 0x7eb10000  0x0 0x1000>,  /* INTC */
+		      <0x0 0x7eb00000  0x0 0x10000>; /* HEVC */
+		reg-names = "intc",
+			    "hevc";
 		interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
-	};
-
-	h264-decoder@7eb20000 {
-		compatible = "raspberrypi,rpivid-h264-decoder";
-		reg = <0x0 0x7eb20000  0x0 0x10000>;
-		status = "okay";
-	};
 
-	vp9-decoder@7eb30000 {
-		compatible = "raspberrypi,rpivid-vp9-decoder";
-		reg = <0x0 0x7eb30000  0x0 0x10000>;
-		status = "okay";
+		clocks = <&firmware_clocks 11>;
+		clock-names = "hevc";
 	};
 };
 
-- 
2.32.0

